src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7716,6 +7718,13 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
 
             _ = MessageID.IDS_FeatureIndexOperator.CheckFeatureAvailability(diagnostics, syntax);
             checkWellKnown(WellKnownMember.System_Index__GetOffset);
+            if (arguments.Names.Count > 0)
+            {
+                diagnostics.Add(
+                    ErrorCode.ERR_ImplicitIndexerWithName,
+                    arguments.Names[0].GetLocation(),
+                    argIsRange ? nameof(Range) : nameof(Index));
```

Comment:
>nameof(Range) : nameof(Index) [](start = 33, length = 29)

It is probably better to have two distinct errors rather than using a string parameter.